docs/notes/04-docstrings.md

```diff
@@ -1,4 +1,60 @@
 # Docstrings
 
-- We'll use [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) (over google style), will need [mkdocstrings](https://mkdocstrings.github.io/griffe/docstrings/#parsers-features) to parse into docs.
-- We use [type annotation](https://docs.python.org/3/library/typing.html)
+## Overview
+
+A docstring, short for documentation string, is a form of inline documentation written as a comment in the source code. It offers a comprehensive description of a module, class, method, or function, intended primarily for developers and advanced users. The typical syntax for a docstring involves enclosing the documentation within triple double quotes as shown below:
+
+```python
+"""This is a docstring.
+
+It can span across multiple lines.
+"""
+```
+
+## Docstring Best Practices
+
+### Docstring Style
+We recommend adhering to a recognized docstring style for consistency and readability. We recommend the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) style (over others like Google or pep257). Following a standard style has several benefits:
+
+1. **Improved Code Readability:** A standard format is familiar to developers, including yourself in the future, allowing for quick access to specific information.
+2. **Integration with Documentation Tools:** Tools like `MkDocs` and `sphinx` can parse standardized docstring formats to automatically generate comprehensive and visually appealing code documentation.
+3. **Consistency:** Adhering to a single docstring style assures consistency across your code base.
+4. **Facilitated Onboarding of New Members:** When a new member joins your team, your docstrings standards will be clear and familiar from the get-go.
+5. **Reduced Risk of Misinterpretation:** With a set standard there will be less ambiguity in your docstrings specifications, reducing the risk of misinterpretation.
+
+### Docstring Linting
+Failing to conform to convention standards may result in improper documentation rendering. As your package grows, these violations may become harder to spot. We therefore strongly suggest using [`pydocstyle`](http://www.pydocstyle.org/en/latest/), a linter that checks the docstrings for you. You can find additional information about linters and code style in the [Linters and Tests](#05-linters-and-tests/) section.
+
+#### Configuration note
+The script `version.py` is autogenerated at build time by `setuptools-scm` (see [packaging note](02-packaging.md)). This script will break the numpy conventions and `pydocstyle` will certainly catch that, resulting in an error! You can address the issue in one of two ways:
+1. **Exclude the `version.py` file:** To ensure `pydocstyle` doesn't analyze this file, modify the match regex in the `pyproject.toml` configuration as shown below:
+    ```toml
+    [tools.pydocstyle]
+    # other configs...
+    match =  "^(?!.*version\\.py$).*\\.py$" # this matches any file ending in .py, except version.py.
```

Comment:
(Removing the `.*` at the beginning should handle make it only match `version.py`, and I think pydocstyle searches all directories recursively), so it would match `version.py` found in any directory within `src`)

Reply:
yes, that worked